### ISSUE TITLE
请问数据处理格式直接是数据的路径 标签吗？在训练但是精度却很低啊，一直都是零点零几？

### ISSUE THREAD

**Issue by DetectionIIT** [wujiyang/Triplet-Loss#10]
mnist/train/5/00000.png 5
mnist/train/0/00001.png 0
mnist/train/4/00002.png 4

以上为数据输入的train.txt文件

**Comment by DetectionIIT** [comment]
三个文件已经在caffe中编译好了后。用的您examples里面的网络先做一个测试。我就修改了数据输入层及输入文件，输入文件如上所示，修改最后有num_output 为我分的21类。结果输出精度一直为0.07左右.学。。请问是不是数据输入格式不对内，没有进行学习内，谢谢
如下为测试结果：
0 1 20 0.05
1 1 20 0.05
2 0 20 0.0
3 0 20 0.0
4 0 20 0.0
5 0 20 0.0
6 2 20 0.1
7 0 20 0.0
8 0 20 0.0
9 0 20 0.0
10 0 20 0.0
11 0 20 0.0
12 0 20 0.0
13 0 20 0.0
14 0 20 0.0
15 0 20 0.0
16 0 20 0.0
17 9 20 0.45
18 4 20 0.2
19 0 20 0.0
20 0 20 0.0
totle_acc: 17 420 0.0404761904762

后来 我在网络里面加了测试层，精度还是很低，请问是不是数据输入错误内？

日志如下：

test_iter: 210
test_interval: 300
base_lr: 0.01
display: 100
max_iter: 30000
lr_policy: "multistep"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
snapshot: 10000
stepvalue: 500
stepvalue: 1500
stepvalue: 15000

I0714 16:12:49.904604 31496 solver.cpp:351] Iteration 9000, Testing net (#0)
I0714 16:13:01.207600 31496 solver.cpp:418]     Test net output #0: accuracy = 0.0714286
I0714 16:13:01.207643 31496 solver.cpp:418]     Test net output #1: loss = 0.0118883 (* 1 = 0.0118883 loss)
I0714 16:13:01.841018 31496 solver.cpp:239] Iteration 9000 (1.04854 iter/s, 95.3703s/100 iters), loss = 0.0210576
I0714 16:13:01.841068 31496 solver.cpp:258]     Train net output #0: loss = 0.0210576 (* 1 = 0.0210576 loss)
I0714 16:13:01.841081 31496 sgd_solver.cpp:112] Iteration 9000, lr = 0.0001
I0714 16:14:25.960108 31496 solver.cpp:239] Iteration 9100 (1.18883 iter/s, 84.1161s/100 iters), loss = 0.0695879
I0714 16:14:25.967063 31496 solver.cpp:258]     Train net output #0: loss = 0.0695879 (* 1 = 0.0695879 loss)
I0714 16:14:25.967097 31496 sgd_solver.cpp:112] Iteration 9100, lr = 0.0001
I0714 16:15:52.791149 31496 solver.cpp:239] Iteration 9200 (1.15178 iter/s, 86.8223s/100 iters), loss = 0.0470989
I0714 16:15:52.792438 31496 solver.cpp:258]     Train net output #0: loss = 0.047099 (* 1 = 0.047099 loss)
I0714 16:15:52.792455 31496 sgd_solver.cpp:112] Iteration 9200, lr = 0.0001
I0714 16:17:17.239606 31496 solver.cpp:351] Iteration 9300, Testing net (#0)
I0714 16:17:28.255017 31496 solver.cpp:418]     Test net output #0: accuracy = 0.0642857
I0714 16:17:28.255098 31496 solver.cpp:418]     Test net output #1: loss = 0.0119613 (* 1 = 0.0119613 loss)
I0714 16:17:28.934146 31496 solver.cpp:239] Iteration 9300 (1.04015 iter/s, 96.1397s/100 iters), loss = 0.0110702
I0714 16:17:28.934247 31496 solver.cpp:258]     Train net output #0: loss = 0.0110703 (* 1 = 0.0110703 loss)
I0714 16:17:28.934278 31496 sgd_solver.cpp:112] Iteration 9300, lr = 0.0001

**Comment by DetectionIIT** [comment]
source: "/media/wujiyang/data/FaceData/train.txt"

您的这个文件格式可以看一下吗？谢谢
